src/CargoOrder.java

```diff
@@ -1,18 +1,32 @@
 package traingame;
 
+import java.util.Arrays;
+import java.util.ArrayList;
 import java.util.Random;
 
-//NOTE: For now, keeping it simple.  Just random.  Might want to re-balance payout later.
-//For example, to care about the distance between pickup and delivery cities.
+// NOTE: For now, keeping it simple.  Just random.  Might want to re-balance payout later.
+// For example, to care about the distance between pickup and delivery cities.
 public record CargoOrder(Product product, City destination, int payment) {
     public static CargoOrder getRandom(City[] cities) {
-        //Chose a random product, city and payout.
+        // Chose a random product, city and payout.
         Random random = new Random();
         Product chosenProduct = Product.getRandom();
         City chosenCity = cities[random.nextInt(cities.length)];
+        Product chosenCityExport = chosenCity.export();
         int chosenPayout = random.nextInt(20,50);
 
-        //Create and return the CargoOrder.
+        // Make sure the product that's chosen is not the destination city's export.
+        if (chosenProduct == chosenCityExport) {
+            // Triggered remove & reroll.
+            ArrayList<Product> productChoices = new ArrayList<>(Arrays.asList(Product.values()));
+            productChoices.remove(chosenCityExport);
+
+            // 2nd and final random call:
+            int chosenIndexOfProduct = (int) (Math.random() * productChoices.size());
+            chosenProduct = productChoices.get(chosenIndexOfProduct);
+        }
```

Comment:
This works, but is unnecessarily complicated. Rather than picking twice, we can just do the guaranteed-to-work pick first. Also, this is the only place that uses Product.getRandom(), so we can change that to our needs e.g. by replacing it with getRandomExcept(Product excluded).

Reply:
I see what you're saying.  Although, while more complex, the current implementation has a fast-and-simple initial call that works most of the time (100 * (products -1) / products) percent of the time.... ex. 96% once we have 25 products.), and if successful, the second, heavier portion doesn't even need to happen (aside from the conditional statement comparison).  Make it a single attempt as you suggest would entail having the heavier operational cost for all calls.

I initially tried to get it working in Product via a getRandomExcept, but kept running into issues with referencing a non-static from a static context.

Reply:
They say "premature optimization is the root of all evil". So when deciding whether to write code that is more readable or more performant, we need to look at exactly how much computation will be saved by being more performant, and whether it is worth it. In this case, it's a few operations per time a CargoOrder is generated, which happens maybe a couple times per turn. That's hardly anything.
If you really want to optimize for performance, you'll need to code both ways and measure the runtime. Beware as the Java compiler may optimize things in ways that make the result you get meaningless.
If you want to avoid copying the list (and I really wouldn't worry about it - memcpy is fast, and the updated garbage collector is good at dealing with extremely short-lived objects), you can also just keep picking a random element even after a first failure, and hope eventually one is valid.

re the getRandomExcept, see my message on discord

Reply:
Good point.  I've updated it to use a simple recursive method instead.  Easy to read, and still fast.  And yea, the time is quite insignificant at ~90k nanoseconds.